nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/test/java/org/nuxeo/ecm/platform/rendition/service/lazy/TestDeliverStaleLazyRenditions.java

```diff
@@ -155,6 +164,35 @@ public void testOnlyStoreUpToDateRenditions() {
         checkUpToDateRendition(true);
     }
 
+    // NXP-27567
+    @Test
+    public void testOnlyConsiderRenditionUpToDateIfBlobsArePutInTransientStore() {
+        // ask for the initial rendition: up-to-date and stored
+        checkInitialRendition(true);
+
+        // get source document modification date transient store parameter
+        RenditionDefinition renditionDefinition = new RenditionDefinition();
+        renditionDefinition.setName(LAZY_AUTOMATION);
+        AutomationLazyRenditionProvider lazyRenditionProvider = new AutomationLazyRenditionProvider();
+        String renditionKey = lazyRenditionProvider.buildRenditionKey(doc, renditionDefinition);
+        TransientStore transientStore = transientStoreService.getStore(CACHE_NAME);
+        String tsSourceDocumentModificationDate = (String) transientStore.getParameter(renditionKey,
+                SOURCE_DOCUMENT_MODIFICATION_DATE_KEY);
+
+        // ask for an intermediate rendition without waiting for its completion: stale and not stored
+        checkStaleRendition(true, false);
+
+        // wait for source document modification date transient store parameter to be updated,
+        // after which the rendition blobs should be put in the transient store
+        String key;
+        do {
+            key = (String) transientStore.getParameter(renditionKey, SOURCE_DOCUMENT_MODIFICATION_DATE_KEY);
+        } while (tsSourceDocumentModificationDate.equals(key));
```

Comment:
How long does this typically take? Could you add a 1ms sleep in the loop to avoid hogging the CPU?

Reply:
I've also added a timeout to avoid the test to freeze if the asynchronous work never ends.